File: An 2/Sem 1/MAP/Lab6/C#/Lab2/Model/UndergraduateStudent.cs
using System;

namespace Lab2
{
	namespace Lab2_Model {
		class UndergraduateStudent : Student, ComparableStudent
		{
			public int grade2;

			public UndergraduateStudent(int id, String name, int grade, int grade2) : base(id, name, grade) {
				this.grade2 = grade2;
			}

			public override float average() {
				return (this.grade + this.grade2) / 2;
			}

			bool ComparableStudent.isGreaterThan(Student student) {
				return (this.average() > student.average());
			}
		}
	}
}

File: An 2/Sem 1/MAP/Lab6/Java/src/lab2/Model/UndergraduateStudent.java
package lab2.Model;

public class UndergraduateStudent extends Student implements ComparableStudent {
    public int grade2;

    public UndergraduateStudent(int id, String name, int grade, int grade2) {
        super(id, name, grade);
        this.grade2 = grade2;
    }

    @Override
    public float average() {
        return (this.grade + this.grade2) / 2;
    }

    @Override
    public boolean isGreaterThan(Student student) {
        return (this.average() > student.average());
    }
}
